proto/substrait/algebra.proto

```diff
@@ -1611,6 +1612,16 @@ message Expression {
   }
 }
 
+// Expression that represents a dynamic parameter.
+// Dynamic parameters are identified by a surrogate key within a plan.
+message DynamicParameter {
+  // The type of the dynamic parameter.
+  Type type = 1;
+
+  // The surrogate key used within a plan to reference a specific parameter binding.
+  uint32 parameter_anchor = 2;
```

Comment:
I believe we call these references. They are references to the anchors. (e.g. function_reference refers to function_anchor).